testcafe/tests/drive_sharing.js

```diff
@@ -0,0 +1,127 @@
+import { Selector, Role } from 'testcafe'
+import { driveUser } from './helpers/roles'
+import {
+  TESTCAFE_DRIVE_URL,
+  isExistingAndVisibile,
+  getCurrentDateTime,
+  FOLDER_DATE_TIME
+} from './helpers/utils'
+import Data from './helpers/data'
+const data = new Data()
+
+import DrivePage from './pages/drive-model'
+const drivePage = new DrivePage()
+
+import PublicDrivePage from './pages/drive-model-public'
+const publicDrivePage = new PublicDrivePage()
+
+const file = '[cozy]QA_table_ APPS.pdf'
+
+//************************
+//Tests when authentified
+//************************
+fixture`Folder link Sharing Scenario`.page`${TESTCAFE_DRIVE_URL}/`.beforeEach(
+  async t => {
+    await t.useRole(driveUser)
+    await drivePage.waitForLoading()
+  }
+)
+
+test('Drive : Create a $test_date_time folder in Drive', async t => {
+  await drivePage.addNewFolder(FOLDER_DATE_TIME)
+  //We need to pass FOLDER_DATE_TIME through multiple fixture, so we cannot use ctx here.
+})
+
+test('Drive : from Drive, go in a folder, upload a file, and share it', async t => {
+  await drivePage.goToFolder(FOLDER_DATE_TIME)
+  await drivePage.openActionMenu()
+  await t.pressKey('esc') //close action Menu
```

Comment:
because of the Action menu overlay, the action menu needs to be close to click on another button outside of it (ie: Share). I choose to use the 'esc' button, but could have just click outside the action menu once, then click on the 'share button'

Reply:
Why do you need to open the ActionMenu ?

Reply:
Because Thomas asked it! ;)
that's part of the scenario 
>     Action : Depuis Drive, je rentre dans le nouveau dossier $test_date_heure
    Résultats attendus :
        le breadcrump a changé, affiche le nom du dossier
        le menu d'action est disponible

Reply:
Le menu d'action c'est pas la cozy-bar. C'est le menu à droite avec les "..."